### ISSUE TITLE
Almalinux build: Error BaseCollection.cc.o

### ISSUE THREAD

**Issue by panagiotisA** [eudaq/eudaq#727]
Trying to build eudaq on almalinux, with root 6.28.06 installed and C++ 17 standard enabled, with cmake variables:

cmake -DUSER_TLU_BUILD=ON -DUSER_CMSIT_BUILD=ON -DEUDAQ_BUILD_STDEVENT_MONITOR=ON -DUSER_CMSPIXEL_BUILD=ON -DCMAKE_INSTALL_PREFIX=../../eudaq_install/ ../

I received the following error

/opt/root-6.28.06_install/include/TFile.h:328:36: note: ‘std::string_view’ is only available from C++17 onwards
make[2]: *** [monitors/onlinemon/CMakeFiles/StdEventMonitor.dir/build.make:87: monitors/onlinemon/CMakeFiles/StdEventMonitor.dir/src/BaseCollection.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1031: monitors/onlinemon/CMakeFiles/StdEventMonitor.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

Any idea how I can resolve this?

**Comment by panagiotisA** [comment]
Yes, the ROOT is installed with C++17, and with the command "root-config --cflags" I receive the following:

-pthread -std=c++17 -m64 -I/opt/root-6.28.06_install/include

**Comment by simonspa** [comment]
Then the problem is your currently available C++ environment. Did you use CVMFS or any other means of getting an up-to-date compiler anc C++ STL? If yes, you need to use the *same* environment to also build programs that depend on ROOT.

**Comment by panagiotisA** [comment]
I am using the gcc 11.3 compiler which it was installed by using (# dnf group install "Development Tools"). The C++17 support is enabled by using  "alias g++='g++ -std=c++17'" on the bashrc. Also when I run the cmake command I get the following error 

"CMake Error: install(EXPORT "eudaqTargets" ...) includes target "module_tlu" which requires target "tlu_static" that is not in any export set."

Without the flags -DUSER_TLU_BUILD=ON  and  -DEUDAQ_BUILD_STDEVENT_MONITOR=ON I don't get any errors.
